controllers/managedocs_controller.go

```diff
@@ -1050,6 +1075,42 @@ func (r *ManagedOCSReconciler) reconcileRookCephOperatorConfig() error {
 	return nil
 }
 
+func (r *ManagedOCSReconciler) reconcileEgressNetworkPolicy() error {
+	_, err := ctrl.CreateOrUpdate(r.ctx, r.Client, r.egressNetworkPolicy, func() error {
+		if err := r.own(r.egressNetworkPolicy); err != nil {
+			return err
+		}
+		desired := templates.EgressNetworkPolicyTemplate.DeepCopy()
+		dmsURL, err := url.Parse(string(r.deadMansSnitchSecret.Data["SNITCH_URL"]))
+		if err != nil {
+			return fmt.Errorf("Unable to parse DMS url: %v", err)
+		}
+		desired.Spec.Egress[0].To.DNSName = dmsURL.Hostname()
+		desired.Spec.Egress[1].To.DNSName = string(r.smtpSecret.Data["host"])
```

Comment:
You cannot use a secret without verifying that it was loaded (as load it if it wasn't). If you do so it might be that you are just looking into an empty object. 

You should also not assume that this will run after or before any other reconcileXXXX function.

```suggestion
		if r.deadMansSnitchSecret.UID == "" {
			if err := r.get(r.deadMansSnitchSecret); err != nil {
				return fmt.Errorf("Unable to get DeadMan's Snitch secret: %v", err)
			}
		}
		dmsURLAsString := string(r.deadMansSnitchSecret.Data["SNITCH_URL"])
		if dmsURL == "" {
			return fmt.Errorf("DeadMan's Snitch secret does not contain a SNITCH_URL entry")
		}
		if dmsURL, err := url.Parse(dmsURLAsString); err == nil {
			desired.Spec.Egress[0].To.DNSName = dmsURL.Hostname()
			desired.Spec.Egress[1].To.DNSName = string(r.smtpSecret.Data["host"])
		} else {
			return fmt.Errorf("Unable to parse DMS url: %v", err)
		}
```

In order to prevent re-get of the secret in case of order change, I will also suggest adding the guard 
```if r.deadMansSnitchSecret.UID == "" {``` 
To the secret get on the `reconcileAlertmanagerConfig` function